benchmark/CMakeLists.txt

```diff
@@ -19,29 +19,29 @@ include(benchmark/dependencies/mpl)
 include(benchmark/dependencies/std)
 
 
-macro(add_dataset datasets dataset path_to_template range)
-    string(REGEX REPLACE "[^.]+[.][^.]+[.][^.]+[.](.*)" "\\1" name ${dataset})
-    set(args ${dataset} ${path_to_template} ${range} NAME ${name} OUTPUT ${name}.json ${ARGN})
+macro(add_dataset datasets type case lib datatype path_to_template range)
+    set(name "${lib}.${datatype}")
+    set(dataset "datasets.${type}.${case}.${name}")
```

Comment:
Also, `type` is not a great name since it's very general. Perhaps `category` would be better?